templates/default.yaml

```diff
@@ -525,6 +525,11 @@ hostResolver:
 # 🟢 Builtin default: /usr/local
 guestInstallPrefix: null
 
+# When "suspend" is enabled, Lima suspends the VM instead of shutting it down.
+# The VM can be resumed with `limactl start` if the saved state is available.
+# 🟢 Builtin default: false
+suspend: null
```

Comment:
Do we need this yaml property?
Can we just have `limactl suspend` command? (Or `limactl stop --suspend`)

Reply:
It’s hard to imagine needing to run `limactl suspend` before restarting the host OS just to take advantage of suspend’s benefits.

Reply:
Adding flags to stop makes it more complicated. I would go with adding suspend/resume commands and keep the ux simple.

You either do:

    limactl stop
    limactl start

Or:

    limactl suspend
    limactl resume

This is also virsh interface (using destroy instead of stop).

Reply:
I delegate the startup and shutdown of my Lima Docker instance to `launchd`, so the Docker instance starts and stops with macOS login and logout. In this PR, I aim to change that behavior to use resume/suspend instead.

For this, I want `hostagent` to suspend when it receives a SIGINT (sent by `limactl stop` or `launchd`) if the `.suspend` option is set to `true`, and to resume on launch if it was previously suspended.  
Adding `limactl (suspend|resume)` would require a separate mechanism to send/receive something other than SIGINT to handle suspension, increasing the complexity of the changes and not aligning with `launchd` shutdown handling. These aren’t changes I’m interested in.

Reply:
> Do we need this yaml property?

Passing it as a command-line option to `limactl start` that would then be forwarded to `hostagent` could work as well. However, I’m also considering setting `.suspend = true` in `override.conf`, so I’d prefer to keep it as an option in `limayaml`.

Reply:
> Adding `limactl (suspend|resume)` would require a separate mechanism to send/receive something other than SIGINT to handle suspension,

Ah, for my request to add `--shut-down` or `--suspend` options to `limactl stop`, communication beyond SIGINT would indeed be required. In that case, I feel that adding `limactl suspend` could be a good solution. 

By the way, what would be the best method to send a request to the hostagent? Should I use a signal other than SIGINT, or a socket?

Reply:
Shouldn't this just use the gRPC communication between guest and hostagent?

Reply:
I don't like the additional `resume` command. I don't care if the VM is suspended or stopped, I should always be able to use `start` to get it running again.

I think SIGINT should suspend a running VM (if possible), and `stop` should stop it (`stop` would no longer send SIGINT, but uses grpc to do it).

Now the only missing functionality would be to suspend a VM by command and not by sending SIGINT. This could be done by a `suspend` subcommand, but I think `stop --suspend` works better, given that we won't have a `resume` command.

Reply:
The communication I'm referring to is the instruction from `limactl suspend` to `limactl hostagent` to trigger suspend.  
`limactl stop` uses SIGINT.  
Currently, I'm considering sending a request with `.suspend = true` to `ha.sock`, followed by sending SIGINT.  
Also, I am thinking that it might be better to name it `.suspendOnStop` instead of `.suspend`.

Reply:
Sorry, I was confused and thought this was between host and guestagent.

Reply:
I plan to try the following implementation:

- Add `.suspend`, `.suspendOnStop`, or `.suspendOnSigInt` to `limayaml`.
- `limactl resume`: Not needed.
- `limactl start`: If the suspend state is still present, resume.
- `limactl suspend`: Request `.suspendOnSigInt = true` to `ha.sock` and send SIGINT to `hostagent`.
- `limactl stop`: Request `.suspendOnSigInt = false` to `ha.sock` and send SIGINT to `hostagent`.
- `hostagent`: Upon receiving SIGINT, if `.suspend` is true, suspend; if false, shutdown.

Reply:
I don't understand why you need the extra field in `limayaml`.

I think I would just use SIGUSR1 to mean `stop` and leave SIGINT to mean `suspend`. That way everything works automatically, including your `launchd` scenario, and you can still explicitly shut down the VM when you want to.

Reply:
On further thoughts I think `limactl stop` shut do a suspend, and `limactl stop --shutdown` (and `limactl stop -f`) should shut down and not suspend.

Eventually we could add a `limactl reboot` (or `restart`) command because I think the only time you really want to shutdown the instance is when you want a fresh boot.

Reply:
That is a good point, I agree.

Reply:
Having an option in the yaml and in the command line sounds like best way:
- If you always want the feature, you use change the default or override yaml
- If you want the feature only for some vms, you enable it in the instance yaml
- If you want the feature only in some cases, you enable it when stopping
- If works like other features, easier to use, easier to maintain